utils/dom.js

```diff
@@ -57,8 +58,13 @@ export function isHorizontalEdge( container, isReverse, collapseRanges = false )
 	}
 
 	const maxOffset = node.nodeType === TEXT_NODE ? node.nodeValue.length : node.childNodes.length;
-
-	if ( ! isReverse && offset !== maxOffset ) {
+	const editor = node.nodeType !== TEXT_NODE && tinymce.get( node.getAttribute( 'id' ) );
```

Comment:
Minor: `id` is a property of `Element`:

```js
node.id
```

https://developer.mozilla.org/en-US/docs/Web/API/Element/id